tests/test_hintfonts.py

```diff
@@ -32,16 +37,42 @@ def test_ufo(ufo, tmpdir):
 
 @pytest.mark.parametrize("otf", glob.glob("%s/*/*/font.otf" % DATA_DIR))
 def test_otf(otf, tmpdir):
-    out = str(tmpdir / basename(otf))
+    out = _temp(tmpdir, otf) + ".out"
     options = Options(otf, out)
     hintFiles(options)
 
     for path in (otf, out):
         font = TTFont(path)
         assert "CFF " in font
-        writer = XMLWriter(path + ".xml")
+        writer = XMLWriter(_temp(tmpdir, path) + ".xml")
         font["CFF "].toXML(writer, font)
         del writer
         del font
 
-    assert differ([otf + ".xml", out + ".xml"])
+    assert differ([_temp(tmpdir, otf) + ".xml", _temp(tmpdir, out) + ".xml"])
+
+
+@pytest.mark.parametrize("cff", glob.glob("%s/*/*/font.cff" % DATA_DIR))
+def test_cff(cff, tmpdir):
+    out = _temp(tmpdir, cff) + ".out"
+    options = Options(cff, out)
+    hintFiles(options)
+
+    for path in (cff, out):
+        font = CFFFontSet()
+        writer = XMLWriter(_temp(tmpdir, path) + ".xml")
+        with open(path, "rb") as fp:
+            font.decompile(fp, None)
+            font.toXML(writer)
+        del writer
+        del font
```

Comment:
is this necessary? aren't they garbage-collected automatically at each iteration, since the variables are reassigned to different objects? or are there circular references hidden somewhere?

Reply:
It is just an unprofiled micro optimization that I kept copying from one test to another…

Reply:
Hmm, dropping this breaks the tests. Something fishy is going on…

Reply:
OK, I need to close `XMLWriter` or written data to file will be truncated.